jOOQ-examples/jOOQ-spring-boot-example/src/main/java/org/jooq/example/spring/repository/BookRepository.java

```diff
@@ -0,0 +1,22 @@
+package org.jooq.example.spring.repository;
+
+import org.jooq.Configuration;
+import org.jooq.DSLContext;
+import org.jooq.example.db.h2.tables.daos.BookDao;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepository extends BookDao {
+
+  private final DSLContext dslContext;
+
+  public BookRepository(Configuration configuration, DSLContext dslContext) {
+    super(configuration);
+    this.dslContext = dslContext;
+  }
+
+  @Override
+  public DSLContext ctx() {
```

Comment:
this one was more tricky

Every time we call the dao-generated method. it creates a new DSLContext instance.

How about removing ```final``` in DefaultConfiguration?
https://github.com/jOOQ/jOOQ/blob/main/jOOQ/src/main/java/org/jooq/impl/DefaultConfiguration.java#L347

it makes other people configure more easy.
chose loading DSL like this.

<img width="550" alt="image" src="https://github.com/jOOQ/jOOQ/assets/50672087/9a1b7ac6-4dee-466d-b058-61aa675d489f">